src/builder/PriceFeeds.sol

```diff
@@ -0,0 +1,284 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.23;
+
+import {Strings} from "./Strings.sol";
+
+library PriceFeeds {
+    error NoPriceFeedPathFound(string inputAssetSymbol, string outputAssetSymbol);
+
+    struct PriceFeed {
+        uint256 chainId;
+        /// @dev The asset symbol for the base currency e.g. "ETH" in ETH/USD
+        string baseSymbol;
+        /// @dev The asset symbol for the quote currency e.g. "USD" in ETH/USD
+        string quoteSymbol;
+        address priceFeed;
+    }
+
+    /// @dev Addresses fetched from: https://docs.chain.link/data-feeds/price-feeds/addresses
+    function knownPriceFeeds() internal pure returns (PriceFeed[] memory) {
+        PriceFeed[] memory mainnetFeeds = knownPriceFeeds_1();
+        PriceFeed[] memory baseFeeds = knownPriceFeeds_8453();
+        PriceFeed[] memory sepoliaFeeds = knownPriceFeeds_11155111();
+        PriceFeed[] memory baseSepoliaFeeds = knownPriceFeeds_84532();
+
+        uint256 totalLength = mainnetFeeds.length + baseFeeds.length + sepoliaFeeds.length + baseSepoliaFeeds.length;
+        PriceFeed[] memory allFeeds = new PriceFeed[](totalLength);
+
+        uint256 currentIndex = 0;
+        for (uint256 i = 0; i < mainnetFeeds.length; ++i) {
+            allFeeds[currentIndex] = mainnetFeeds[i];
+            currentIndex++;
+        }
+        for (uint256 i = 0; i < baseFeeds.length; ++i) {
+            allFeeds[currentIndex] = baseFeeds[i];
+            currentIndex++;
+        }
+        for (uint256 i = 0; i < sepoliaFeeds.length; ++i) {
+            allFeeds[currentIndex] = sepoliaFeeds[i];
+            currentIndex++;
+        }
+        for (uint256 i = 0; i < baseSepoliaFeeds.length; ++i) {
+            allFeeds[currentIndex] = baseSepoliaFeeds[i];
+            currentIndex++;
+        }
+
+        return allFeeds;
+    }
+
+    // Mainnet
+    function knownPriceFeeds_1() internal pure returns (PriceFeed[] memory) {
+        PriceFeed[] memory priceFeeds = new PriceFeed[](4);
```

Comment:
Do we plan to add every price feed for all supported tokens here?

Reply:
Yeah, we'll have to hardcode all the price feeds for tokens we care about. 🧇 is working on a crawler that will output a Solidity config file that will replace this file.